ext/storage_sources/gcp_store/test/test_gcp_connection.cpp

```diff
@@ -30,5 +30,338 @@
+ * pattern or lifecycle of the object, please speak to the release manager.
+ */
+static std::string obj_prefix("gcptest"); // To be concatenated with a random string.
+} // namespace test_defaults
+
+// Concatenates a random suffix to the prefix being used for the test object keys. Example of
+// generated test prefix: "gcptest/unit/2022-31-01-16-34-10/623843294--".
+static int
+randomize_test_prefix()
+{
+    char time_str[100];
+    std::time_t t = std::time(nullptr);
+    std::string new_prefix = "gcptest-unit-";
+
+    REQUIRE(std::strftime(time_str, sizeof(time_str), "%F-%H-%M-%S", std::localtime(&t)) != 0);
+
+    new_prefix += time_str;
+
+    // Create a random device and use it to generate a random seed to initialize the generator.
+    std::random_device my_random_device;
+    unsigned seed = my_random_device();
+    std::default_random_engine my_random_engine(seed);
+
+    new_prefix += '/' + std::to_string(my_random_engine());
+    new_prefix += "--";
+
+    test_defaults::obj_prefix = new_prefix;
+
+    return (0);
+}
+
+// Overrides the defaults with the ones specific for this test instance.
+static int
+setup_test_defaults()
+{
+    // Append the prefix to be used for object names by a unique string.
+    REQUIRE(randomize_test_prefix() == 0);
+    // std::cerr << "Generated prefix: " << test_defaults::obj_prefix << std::endl;
+
+    return (0);
+}
+
+TEST_CASE("Testing class gcpConnection", "gcp-connection")
+{
+    // Setup the test environment.
+    REQUIRE(setup_test_defaults() == 0);
+
+    gcp_connection conn(test_defaults::bucket_name, test_defaults::obj_prefix);
+    // bool exists = false;
+    // size_t objectSize;
+
+    const std::string object_key = "test_object";
+    const std::string file_name = "test_object.txt";
+    const std::string file_path = "./" + file_name;
+    const std::string non_exi_object_key = "test_non_exist";
+    const std::string non_exi_file_name = "test_non_exist.txt";
+    const bool list_single = true;
+    std::vector<std::string> objects;
+
+    google::cloud::storage::Client client = conn.get_client();
+
+    std::ofstream File(file_name);
+    std::string payload = "Test payload :)";
+    File << payload;
+    File.close();
+
+    SECTION("Simple list test", "[gcp-connection]")
+    {
+        const int32_t total_objects = 20;
+
+        // No matching objects. Object size should be 0.
+        REQUIRE(conn.list_objects(objects, false) == 0);
+        REQUIRE(objects.size() == 0);
+
+        // No matching objects with list_single. Object size should be 0.
+        REQUIRE(conn.list_objects(objects, list_single) == 0);
+        REQUIRE(objects.size() == 0);
+
+        // Upload 1 file and test list.
+        objects.clear();
+        client.UploadFile(
+          file_path, test_defaults::bucket_name, test_defaults::obj_prefix + object_key);
+        REQUIRE(conn.list_objects(objects, false) == 0);
+        REQUIRE(objects.size() == 1);
+
+        // Delete that object and test list.
```

Comment:
We could improve this comment by describing the intent and the expected outcome of the test.

Reply:
addressed here: [d14ba2e](https://github.com/wiredtiger/wiredtiger/pull/8749/commits/d14ba2e58bd216c13c752390b0921dbbb41f03ed)